src/objects.rs

```diff
@@ -47,18 +55,34 @@ pub struct BlueprintBookBlueprintValue {
     pub item: Container,
 }
 
+/// https://wiki.factorio.com/Blueprint_string_format#Blueprint_object
 #[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
 #[serde(default)]
-/// https://wiki.factorio.com/Blueprint_string_format#Blueprint_object
 pub struct Blueprint {
     pub item: String,
-    pub label: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub label: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub label_color: Option<Color>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub entities: Vec<Entity>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub tiles: Vec<Tile>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub icons: Vec<Icon>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub schedules: Vec<Schedule>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "position-relative-to-grid")]
+    pub position_relative_to_grid: Option<Position>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "snap-to-grid")]
```

Comment:
Why use individual renames on fields? It would be more efficient to do the renames per container:

```rust
#[serde(default, rename_all="kebab-case")]
pub struct Blueprint {
```

Reply:
I don't recall exactly but looking at it now, I see label_color is not kebab-case, which makes me believe that's the reason I didn't. At this point more are kebab-case so making container rename_all and then snake_casing label_color if it needs it is probably better.